### PR TITLE
 implement VTimestampField and VWeekField

### PR DESCRIPTION
**Pull request by SNsir**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/form/VDateField.kt [hunk 1]
```diff
@@ -216,7 +216,7 @@ class VDateField : VField(10, 1) {
     return value[r]!!
   }
 
-  override fun toText(o: Any): String {
+  override fun toText(o: Any?): String {
     return if (o == null) "" else Companion.toText(o as Date)
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VDateField.kt:220]: remove Companion.

File: galite-core/src/main/kotlin/org/kopi/galite/form/VWeekField.kt [hunk 2]
```diff
@@ -15,6 +15,455 @@
...
+      parseWeek(rec, s)
+    }
+  }
+
+  private fun parseWeek(rec: Int, s: String) {
+    var week = 0
+    var year = -1
+    var bp = 0
+    var state: Int
+    val buffer = s + '\u0000'
```
[the remark this answers is not in the document]
Comment (SNsir) [reply galite-core/src/main/kotlin/org/kopi/galite/form/VWeekField.kt:100]: they are correct

File: galite-core/src/main/kotlin/org/kopi/galite/form/VColorField.kt [hunk 3]
```diff
@@ -150,7 +150,7 @@ class VColorField(width: Int, height: Int) : VField(1, 1) {
   override fun getObjectImpl(r: Int): Any? = value[r]
 
   override fun toText(o: Any?): String {
-    throw InconsistencyException("UNEXPECTED GET TEXT")
+    throw InconsistencyException("UNEXPECTD GET TEXT")
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VColorField.kt]: UNEXPECTED

File: galite-core/src/main/kotlin/org/kopi/galite/form/VDateField.kt [hunk 4]
```diff
@@ -217,7 +217,7 @@ class VDateField : VField(10, 1) {
   }
 
   override fun toText(o: Any?): String {
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VDateField.kt]: inline function

**Review by h-haddad — APPROVED** [review]

**Pull request merged**